src/pages/workspace/WorkspaceSettingsCurrencyPage.js

```diff
@@ -0,0 +1,102 @@
+import React, {useState, useCallback} from 'react';
+import _ from 'underscore';
+import {withOnyx} from 'react-native-onyx';
+import PropTypes from 'prop-types';
+import useLocalize from '../../hooks/useLocalize';
+import ScreenWrapper from '../../components/ScreenWrapper';
+import HeaderWithBackButton from '../../components/HeaderWithBackButton';
+import SelectionList from '../../components/SelectionList';
+import Navigation from '../../libs/Navigation/Navigation';
+import ROUTES from '../../ROUTES';
+import compose from '../../libs/compose';
+import ONYXKEYS from '../../ONYXKEYS';
+import withPolicy, {policyDefaultProps, policyPropTypes} from './withPolicy';
+import * as Policy from '../../libs/actions/Policy';
+
+const propTypes = {
+    /** Constant, list of available currencies */
+    currencyList: PropTypes.objectOf(
+        PropTypes.shape({
+            /** Symbol of the currency */
+            symbol: PropTypes.string.isRequired,
+        }),
+    ),
+    ...policyPropTypes,
+};
+
+const defaultProps = {
+    currencyList: {},
+    ...policyDefaultProps,
+};
+
+const getDisplayText = (currencyCode, currencySymbol) => `${currencyCode} - ${currencySymbol}`;
+
+function WorkspaceSettingsCurrencyPage({currencyList, policy}) {
+    const {translate} = useLocalize();
+    const [searchText, setSearchText] = useState('');
+    const trimmedText = searchText.trim().toLowerCase();
+    const currencyListKeys = _.keys(currencyList);
+
+    const filteredItems = _.filter(currencyListKeys, (currencyCode) => {
+        const currency = currencyList[currencyCode];
+        return getDisplayText(currencyCode, currency.symbol).toLowerCase().includes(trimmedText);
+    });
+
+    let initiallyFocusedOptionKey;
+
+    const currencyItems = _.map(filteredItems, (currencyCode) => {
+        const currency = currencyList[currencyCode];
+        const isSelected = policy.outputCurrency === currencyCode;
+
+        if (isSelected) {
+            initiallyFocusedOptionKey = currencyCode;
+        }
+
+        return {
+            text: getDisplayText(currencyCode, currency.symbol),
+            keyForList: currencyCode,
+            isSelected,
+        };
+    });
```

Comment:
Maybe we could memoize this?

Reply:
It was memoized previously, but we decided to remove it as it had no effect, since it had every state as a dependency